pkg/apis/flyteworkflow/v1alpha1/node_status.go

```diff
@@ -523,7 +523,7 @@ func (in *NodeStatus) GetNodeExecutionStatus(ctx context.Context, id NodeID) Exe
 		n.SetParentTaskID(in.GetParentTaskID())
 		n.DataReferenceConstructor = in.DataReferenceConstructor
 		if len(n.GetDataDir()) == 0 {
-			dataDir, err := in.DataReferenceConstructor.ConstructReference(ctx, in.GetOutputDir(), id)
+			dataDir, err := in.DataReferenceConstructor.ConstructReference(ctx, in.GetDataDir(), id)
```

Comment:
Shouldn't it be
```suggestion
			dataDir, err := in.DataReferenceConstructor.ConstructReference(ctx, in.GetOutputDir(), id)
```

Reply:
I tried https://github.com/flyteorg/flytepropeller/pull/292/commits/8310d4fb822c5f97b90216f3209f55ccf5d788e3 in lyft staging and it fixed the issue

Reply:
Yeah it's just this is "magic" code that I don't want to have around... super hard to maintain... so let's revert this and understand the issue and fix it properly...